pkg/operator/ceph/config/config.go

```diff
@@ -133,6 +133,57 @@ func SetDefaultAndUserConfigs(
 	return nil
 }
 
+// SetOneTimeDefaults sets default values for Ceph that should only be set once upon cluster
+// creation.
+func SetOneTimeDefaults(
+	context *clusterd.Context,
+	namespace string,
+	clusterInfo *cephconfig.ClusterInfo,
+	configOverrides rookceph.ConfigOverridesSpec,
+) error {
+	// ceph.conf is never used. All configurations are made in the centralized mon config database,
+	// or they are specified on the commandline when daemons are called.
+	monStore := GetMonStore(context, namespace)
+
+	// one-time defaults
+	defaults := map[string]string{
+		"osd pool default pg num":  "128",
+		"osd pool default pgp num": "128",
```

Comment:
There are some hints about selecting a pg number in ceph docs: [A preselection of pg_num](https://docs.ceph.com/docs/master/rados/operations/placement-groups/#a-preselection-of-pg-num).

And it seems that there may be a simple answer to the pg number, but for small clusters only:

> If you have more than 50 OSDs, you need to understand the tradeoffs and how to calculate the pg_num value by yourself

But even for small clusters, pg num set to 100 is not a good one:

> pg number should always be rounded up to the nearest power of two

Reply:
IIRC Octopus should enable the PG autocaler by default on new clusters. I'm **really** against changing Ceph's default values of `osd_pool_default_pg_num` and `osd_pool_default_pgp_num`. I believe, by now, it has been well understood that PGs must be configured appropriately according to your cluster design (amount of OSDs). Changing the value to 128 is not helping.

Our CI should set both values in accordance to the CI cluster test so if 128 is enough then configure the Test suite to deploy a cluster with 128. 

For everything else, user should be aware that this has to be configured. So this is mostly doc for me.

To sum up, I'm against re-introducing old values, as well as adding new once. I think what we have now is correct.

Reply:
@leseb @mbukatov @liewegas Can we enable the PG autoscaler for new clusters now? Or do we need to wait for octopus? 

If we do enable it by default now, I would suggest upgraded clusters by default not have it enabled. They could enable a setting in the CRD later when they are ready to enable it.

But if we shouldn't enable the autoscaler until octopus, it seems safer to keep the defaults Rook already has been using rather than introduce new defaults.

Reply:
I'd wait for whatever Ceph does for new clusters but maybe @liewegas has a different option.

I don't think we need a CRD option for that, they can simply go on the toolbox, what'd be nice though is to have the ability to configure MGR modules via the CRD but that's a different discussion.

I still see re-introducing "wrong" values as invalid, even though it hasn't broken anything AFAIK, but since this is the master branch I wouldn't do anything as mentioned above.